src/main/kotlin/de/smartsquare/starter/mqttadmin/emqx/EmqxApiConfiguration.kt

```diff
@@ -1,27 +1,40 @@
 package de.smartsquare.starter.mqttadmin.emqx
 
+import com.fasterxml.jackson.databind.DeserializationFeature
 import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import de.smartsquare.starter.mqttadmin.client.BrokerApiClient
 import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.boot.web.client.RestTemplateBuilder
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
 import org.springframework.web.client.RestTemplate
 
 @Configuration
+@ConditionalOnProperty(prefix = "emqx.api", value = ["host"])
 @EnableConfigurationProperties(EmqxApiProperties::class)
-open class EmqxApiConfiguration {
+class EmqxApiConfiguration {
 
     @Bean
     @Qualifier("emqx")
```

Comment:
Is the qualifier still needed?

Reply:
Good point. I think we can remove the internally used beans altogether.